File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        int mi=nums[0];
        int ma=-1;
        for(int i=1;i<nums.size();i++){
            if(nums[i]>mi) ma=max(ma,nums[i]-mi);
            else mi=min(mi,nums[i]);
        }
        return ma;
    }
};

File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int min=nums[0];
        int max=-1;
        for(int i=1;i<nums.length;i++){
            if(nums[i]>min) max=Math.max(max,nums[i]-min);
            else min=Math.min(min,nums[i]);
        }
        return max;
    }
}